src/snowflake/snowpark/modin/plugin/extensions/base_overrides.py

```diff
@@ -6,31 +6,152 @@
 Methods defined on BasePandasDataset that are overridden in Snowpark pandas. Adding a method to this file
 should be done with discretion, and only when relevant changes cannot be made to the query compiler or
 upstream frontend to accommodate Snowpark pandas.
+
+If you must override a method in this file, please add a comment describing why it must be overridden,
+and if possible, whether this can be reconciled with upstream Modin.
 """
 from __future__ import annotations
 
 import pickle as pkl
-from typing import Any
+import warnings
+from collections.abc import Sequence
+from typing import Any, Callable, Hashable, Literal, Mapping, cast, get_args
 
+import modin.pandas as pd
 import numpy as np
+import numpy.typing as npt
 import pandas
 from modin.pandas.base import BasePandasDataset
-from pandas._libs.lib import no_default
+from pandas._libs import lib
+from pandas._libs.lib import NoDefault, is_bool, no_default
 from pandas._typing import (
+    AggFuncType,
+    AnyArrayLike,
+    Axes,
     Axis,
     CompressionOptions,
+    FillnaOptions,
+    IgnoreRaise,
+    IndexKeyFunc,
+    IndexLabel,
+    Level,
+    NaPosition,
+    RandomState,
+    Scalar,
     StorageOptions,
     TimedeltaConvertibleTypes,
+    TimestampConvertibleTypes,
+)
+from pandas.core.common import apply_if_callable
+from pandas.core.dtypes.common import (
+    is_dict_like,
+    is_dtype_equal,
+    is_list_like,
+    is_numeric_dtype,
+    pandas_dtype,
+)
+from pandas.core.dtypes.inference import is_integer
+from pandas.core.methods.describe import _refine_percentiles
+from pandas.errors import SpecificationError
+from pandas.util._validators import (
+    validate_ascending,
+    validate_bool_kwarg,
+    validate_percentile,
 )
 
+import snowflake.snowpark.modin.pandas as spd
 from snowflake.snowpark.modin.pandas.api.extensions import (
     register_dataframe_accessor,
     register_series_accessor,
 )
+from snowflake.snowpark.modin.pandas.utils import (
+    ensure_index,
+    extract_validate_and_try_convert_named_aggs_from_kwargs,
+    get_as_shape_compatible_dataframe_or_series,
+    is_scalar,
+    raise_if_native_pandas_objects,
+    validate_and_try_convert_agg_func_arg_func_to_str,
+)
 from snowflake.snowpark.modin.plugin._internal.telemetry import (
+    TELEMETRY_PRIVATE_METHODS,
+    PropertyMethodType,
     snowpark_pandas_telemetry_method_decorator,
 )
-from snowflake.snowpark.modin.plugin.utils.error_message import base_not_implemented
+from snowflake.snowpark.modin.plugin._typing import ListLike
+from snowflake.snowpark.modin.plugin.utils.error_message import (
+    ErrorMessage,
+    base_not_implemented,
+)
+from snowflake.snowpark.modin.plugin.utils.warning_message import WarningMessage
+from snowflake.snowpark.modin.utils import validate_int_kwarg
+
+
+def register_base_override(method_name: str):
+    def decorator(base_method: Any):
```

Comment:
what is this function try to do here ? can yo add some comment?

Reply:
Upstream modin doesn't have an equivalent of `register_dataframe_override` for BasePandasDataset, so this method tries to simulate that behavior. I added a comment to that effect.